json_serializable/lib/src/utils.dart

```diff
@@ -229,3 +230,14 @@ extension ExecutableElementExtension on ExecutableElement {
     );
   }
 }
+
+// Return the prefix used by json_annotation if there is one
+String? getPrefixOfJsonAnnotationImport(TypeHelperContextWithConfig context) {
```

Comment:
Why not return `''` – empty string when there is no prefix and add a `.` when there is.

Then you can drop all of the if checks all over.

Maybe?

Reply:
Yes, you re right it will be better !